From 6cb206803fc72933227e4deba19c7d0c9c585d8b Mon Sep 17 00:00:00 2001
From: Chris Blake <chrisrblake93@gmail.com>
Date: Fri, 22 Feb 2019 13:15:00 -0600
Subject: [PATCH 6/9] mmc: sunxi: Remove calibration support for H3/H5

As mentioned at
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?id=d49d92ace4974510dd5845ff91860823cf30dc08,
the H3/H5 SoC does not work with calibration on some boards. Let's
disable it for this SoC.
---
 drivers/mmc/sunxi_mmc.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/sunxi_mmc.c b/drivers/mmc/sunxi_mmc.c
index 9f426661c4..623fb4ce28 100644
--- a/drivers/mmc/sunxi_mmc.c
+++ b/drivers/mmc/sunxi_mmc.c
@@ -120,7 +120,8 @@ static int mmc_set_mod_clk(struct sunxi_mmc_priv *priv, unsigned int hz)
 	if (IS_ENABLED(CONFIG_MACH_SUN8I_A83T) && priv->mmc_no != 2)
 		new_mode = false;
 
-#if defined(CONFIG_MACH_SUN50I) || defined(CONFIG_MACH_SUN50I_H6)
+	/* H3/H5 boards have issues calibrating, so fallback to hard coded values */
+#if !defined(CONFIG_MACH_SUNXI_H3_H5) && (defined(CONFIG_MACH_SUN50I) || defined(CONFIG_MACH_SUN50I_H6))
 	calibrate = true;
 #endif
 
-- 
2.17.1

